packages/ui/src/components/timeline/RangeSelector.tsx

```diff
@@ -51,24 +54,22 @@ export function RangeSelector({rangeRef}: RangeSelectorProps) {
   return (
     <div
       ref={rangeRef}
-      style={{
-        flexDirection: start > end ? 'row-reverse' : 'row',
-        left: `${framesToPercents(Math.ceil(Math.max(0, normalizedStart)))}%`,
-        right: `${
-          100 - framesToPercents(Math.ceil(Math.min(duration, normalizedEnd)))
-        }%`,
-      }}
-      className={clsx(styles.range, shiftHeld && styles.active)}
+      className={clsx(
+        styles.rangeContainer,
+        shiftHeld && controlHeld && styles.active,
+      )}
       onPointerDown={event => {
         if (event.button === MouseButton.Left) {
           event.preventDefault();
           event.stopPropagation();
           event.currentTarget.setPointerCapture(event.pointerId);
+
+          setStart(pixelsToFrames(event.clientX - rect.x));
+          setEnd(pixelsToFrames(event.clientX - rect.x));
```

Comment:
Mapping from the pointer's position to the corresponding animation frame is common logic between the playhead and this. I'd suggest we add a function for this to `TimelineState`. This could be done in `Timeline.tsx` when the state is created:
```ts
const startPosition = sizes.paddingLeft + rect.x - offset;
const pointerToFrames: (value: number) =>
  conversion.pixelsToFrames(value - startPosition),
```
And then used both here and in the playhead scrubbing function.
It would also remove the need for another `useSize` in this component.